scripts/install.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+#
+#  Copyright (c) 2024 Red Hat, Inc.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  You may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Script to handle CLI helm installation for k8s and OCP
+#
+# Requires: oc or kubectl
+
+usage() {
+echo "
+This script simplifies and automates the installation process of Helm charts on Kubernetes (K8s) and OpenShift Container Platform (OCP) clusters.
+It detects whether 'oc' or 'kubectl' is installed and ensures that the user is logged into a cluster.
+The script also attempts to detect the cluster router base and updates the Helm chart configuration accordingly.
+
+Usage:
+  $0 [OPTIONS]
+
+Options:
+  --cli <oc|kubectl>          : Specify the CLI tool to use (overrides auto-detection).
+  --router-base <router-base> : Manually provide the cluster router base if auto-detection fails.
+  --release-name <name>       : Specify a custom release name for the Helm chart.
```

Comment:
In my helm-for-ocp script, I hardcode name to `redhat-developer-hub` and then the namespace defaults to `namespace="rhdh-helm"`

https://github.com/rhdh-bot/openshift-helm-charts/blob/rhdh-1-rhel-9/installation/install.sh

so we end up with this route: https://redhat-developer-hub-${namespace}.${CLUSTER_ROUTER_BASE}

Here we could get all manner of options using <name>.<namespace>.<router-base>.

I wonder if there's value in setting defaults in case people fail to pass in overrides so we can default to things we know work...

Reply:
It should never matter what name and namespace are used, everything should be routed correctly automatically. If someone wants to install multiple developer hub instances on the same cluster, they would currently have to go in and modify the script if we hard code the values, it would make more sense to have flags that they can use to provide the name and namespace I think.